src/report.ts

```diff
@@ -83,7 +115,20 @@ async function pollReport(
 
 export async function reportBundle(options: UploadReportOptions): Promise<ReportResult> {
   // Call remote bundle for analysis results and emit intermediate progress
-  const response = await pollReport(options);
+  const response = await initAndPollReport(options);
+
+  if (response.type === 'error') {
+    throw response.error;
+  } else if (response.value.status === AnalysisStatus.failed) {
+    throw new Error('Analysis has failed');
```

Comment:
q: How do we track analysis failure in this case? We log it somewhere upstream?

Reply:
Yeah it will bubble up and be handled by the CLI logic which calls these methods.